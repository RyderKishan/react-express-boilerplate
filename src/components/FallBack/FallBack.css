.FallBack {
  padding: 32px;
}

.FallBackHeader {
  height: 70px;
  font-size: 18px;
  font-weight: bold;
  line-height: 70px;
}
