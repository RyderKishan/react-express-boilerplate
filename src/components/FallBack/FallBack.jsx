import React from 'react';
import PropTypes from 'prop-types';
import * as qs from 'qs';
import './FallBack.css';

const failureMessages = {
  NOT_FOUND: 'Particular content is not found',
  LOGGED_OUT: 'You have been logged out successfully',
  SESSION_EXPIRED: 'Your session is expired! Please reload',
};

const getErrorMessage = (location) => {
  const params = qs.parse(location.search, { ignoreQueryPrefix: true });
  if (params.error) {
    const failureMessage = failureMessages[params.error];
    if (failureMessage) {
      return {
        text: failureMessage,
        code: `ERRCODE: ${params.error}`,
      };
    }
  }
  return {
    text: failureMessages.NOT_FOUND,
    code: 'ERRCODE: 404',
  };
};

const FallBack = ({ location }) => {
  const message = getErrorMessage(location);
  return (
    <div className="FallBack">
      <div className="FallBackHeader">{message.code}</div>
      <div className="FallBackText">{message.text}</div>
    </div>
  );
};

FallBack.propTypes = {
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }).isRequired,
};

export default FallBack;
