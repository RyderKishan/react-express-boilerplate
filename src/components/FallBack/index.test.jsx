/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import FallBack from '.';

const props = {
  location: {
    search: '?error=LOGGED_OUT',
  },
};

describe('<FallBack /> - Snapshot Tests Renderer', () => {
  const tree = renderer.create(<FallBack {...props} />);
  it('Snapshot renderer', () => {
    expect(tree).toMatchSnapshot();
  });
});

describe('<FallBack /> - Route Tests', () => {
  it('404 Error', () => {
    const localProps = { location: { search: '' } };
    const wrapper = shallow(<FallBack {...localProps} />);
    expect(wrapper.find('.FallBackHeader').text()).toBe('ERRCODE: 404');
  });
  it('Possible Error', () => {
    const localProps = { location: { search: '?error=LOGGED_OUT' } };
    const wrapper = shallow(<FallBack {...localProps} />);
    expect(wrapper.find('.FallBackHeader').text()).toBe('ERRCODE: LOGGED_OUT');
  });
  it('Im-Possible Error', () => {
    const localProps = { location: { search: '?error=ERROR' } };
    const wrapper = shallow(<FallBack {...localProps} />);
    expect(wrapper.find('.FallBackHeader').text()).toBe('ERRCODE: 404');
  });
});
