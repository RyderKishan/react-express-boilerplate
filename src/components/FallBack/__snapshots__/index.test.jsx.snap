// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FallBack /> - Snapshot Tests Renderer Snapshot renderer 1`] = `
<div
  className="FallBack"
>
  <div
    className="FallBackHeader"
  >
    ERRCODE: LOGGED_OUT
  </div>
  <div
    className="FallBackText"
  >
    You have been logged out successfully
  </div>
</div>
`;
